src/tools/miri/tests/fail/abort-terminator.rs

```diff
@@ -1,7 +1,7 @@
 #![feature(c_unwind)]
 
 extern "C" fn panic_abort() {
-    //~^ ERROR: the program aborted
+    //~^ ERROR: panic in a function that cannot unwind
```

Comment:
This test was supposed to test the `Abort` terminator (now called `Terminate`). It no longer does that, so that's a bug.

From the description of the `Terminate` terminator, I also would expect that terminator to be present here: "Used to prevent unwinding for foreign items or with `-C unwind=abort`". Does that comment need to be updated?

Please make sure that we have a test that actually involves the `Terminate` terminator, and another one for "panic in a function that cannot unwind".

Reply:
It's a bit hard, because we currently do not construct `Abort/Terminate` terminator directly anymore. They are only introduced in transforms. IIRC MIRI don't enable MIR optimisations?

Reply:
Oh, okay -- what kind of transforms? Definitely sounds like the docs on the MIR syntax file must be updated then.

Reply:
Inlining will introduce it. When the callee of an `UnwindAction::Terminate` call has cleanup code, the `Resume` terminator of the inlined code will be replaced with `Terminate`. The original version of this PR attempts to get rid of `Terminate` terminator at once by forcing callee to be nounwind as well but that received some pushbacks (see previous discussion) so `Terminate` is kept.

Currently drop elaboration of `DropAndReplace` will create `Terminate` as well but I am looking at how to remove it.

Reply:
It should be possible to have a Miri test for that, with `-Zmir-opt-level=3`. (Miri sets it to 0 by default but that can be overridden.)

Reply:
Looks like `compiler/rustc_mir_transform/src/abort_unwinding_calls.rs` is another pass that can introduce this terminator? (That's the only one I was able to find via grepping.)

Reply:
Abort unwinding calls introduces `UnwindAction::Terminate`, not `TerminatorKind::Terminate`.

Reply:
Ah right, this PR changes that, nice.

Reply:
The new test looks great, thanks. :)